tests/net/socket/tls/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+common:
+  tags: net socket tls
+  filter: TOOLCHAIN_HAS_NEWLIB == 1
+tests:
+  net.socket.tls:
```

Comment:
Not sure if we want to have here two test runs, one for userspace and one for kernel mode. The `tests/net/socket/tcp` test only has userspace test for some reason.

Reply:
I might skip userspace (at first at least) to mainly concentrate on TLS. I'm still a bit confused now & then with object permissions, and particularly with sockets that can be even more confusing.

Reply:
First step is to get things to work in kernel mode. Adding userspace support should not be too complicated. Anyway, support for it can be added later.